### PR TITLE
Add support for useTheme hook

### PR DESCRIPTION
**Pull request by HenriBeck**
As hooks are now in a stable release, I added a small util function for getting the theme inside functional components.

### CODE REVIEW

File: src/create-use-theme.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// @flow
+
+import React, { type Context } from 'react';
+import warning from 'tiny-warning';
+import isObject from './is-object';
+
+export default function createUseTheme<Theme>(context: Context<Theme>) {
```
Comment (TrySound) [inline src/create-use-theme.js]: I'd all it initialValue

File: src/create-use-theme.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  return () => {
```
Comment (TrySound) [inline src/create-use-theme.js]: Make it named please

File: src/create-use-theme.js (hunk 1 above)
Comment (HenriBeck) [reply src/create-use-theme.js]: what, the context?

Comment (TrySound) [reply src/create-use-theme.js]: Ah, I see. It is context

**Comment by kof** [comment]
Missing the docs

File: README.md [hunk 3]
```diff
@@ -218,6 +219,25 @@ export default App;
 
 The `withTheme` HOC supports the new React forwardRef API so you can just use the normal ref prop.
 
+### Use theme (Hook)
```
Comment (kof) [inline README.md]: just call it `useTheme` so that it matches the  hash in TOC

**Comment by HenriBeck** [comment]
Alright, updated the docs

**Pull request merged**